draft-ietf-quic-recovery.md

```diff
@@ -287,8 +322,41 @@ Pseudocode for OnPacketSent follows:
 
 ~~~
  OnPacketSent(packet_number, is_retransmittable):
-     if is_retransmittable:
-       SetLossDetectionAlarm()
+   # TODO: Clarify the data in sent_packets.
+   sent_packets[packet_number] = {now}
+   if is_retransmittable:
+     SetLossDetectionAlarm()
+~~~
+
+## On Ack Receipt
+
+When an ack is received, it may acknowledge 0 or more packets.
+
+Pseudocode for OnAckReceived and UpdateRtt follow:
+
+~~~
+   OnAckReceived(ack):
+     // If the largest acked is newly acked, update the RTT.
+     if (sent_packets[ack.largest_acked]):
+       rtt_sample = now - sent_packets[ack.largest_acked]
+       if (rtt_sample > ack.ack_delay):
+         rtt_sample -= ack.delay;
+       UpdateRtt(rtt_sample)
+     // Find all newly acked packets.
+     for acked_packet in DetermineNewlyAckedPackets():
+       OnPacketAcked(acked_packet)
+
+     DetectLostPackets(ack.largest_acked_packet);
+     SetLossDetectionAlarm();
+
+
+   UpdateRtt(rtt_sample):
+     if (smoothed_rtt == 0):
+       smoothed_rtt = rtt_sample
+       rttvar = rtt_sample / 2
+     else:
+       rttvar = 3/4 * rttvar + 1/4 * (smoothed_rtt - rtt_sample)
+       smoothed_rtt = 7/8 * smoothed_rtt + 1/8 * rtt_sample
```

Comment:
Oh, I see that this is just RFC 6298 directly implemented.  Might pay to be more direct about that.  Only the smoothed_rtt value contains the reference.